client/src/main/scala/org/http4s/client/WebSocketClient.scala

```diff
@@ -0,0 +1,33 @@
+package org.http4s
+package client
+
+import fs2.{Sink, Stream}
+import java.net.SocketAddress
+import org.http4s.websocket.WebsocketBits.WebSocketFrame
+
+trait WebSocketClient[F[_]] {
+  def connect(req: Request[F]): F[WebSocket[F]]
+}
+
+object WebSocketClient {
+  def apply[F[_]](f: Request[F] => F[WebSocket[F]]): WebSocketClient[F] =
+    new WebSocketClient[F] {
+      def connect(req: Request[F]) = f(req)
+    }
+}
+
+abstract class WebSocket[F[_]] {
```

Comment:
This needs to be unified with the one in core.

Cue standard trait vs. abstract class debate.